app/assets/javascripts/host_edit.js

```diff
@@ -221,23 +221,45 @@ function hostgroup_changed(element) {
   var host_id = $("form").data('id');
   var host_changed = $("form").data('type-changed');
   if (host_id) {
-    if (host_changed ){
-      update_form(element,{data:"&host[id]="+host_id});
-    } else if (host_changed == undefined) { // hostgroup changes parent
-      update_form(element);
-    } else { // edit host
-      set_inherited_value(element);
-      update_puppetclasses(element);
-      reload_host_params();
-    }
+    handleHostgroupChangeEdit(element, host_id, host_changed)
   } else { // a new host
-    reset_explicit_values(element);
+    disableComputeResourceDropdown(true);
+    handleHostgroupChangedNew(element).then(function () {
+      disableComputeResourceDropdown(false);
+    });
+  }
+}
+
+function promise() {
```

Comment:
If I understand correctly, you are creating an empty promise to return in case `update_form` is not called in `handleHostgroupChangedNew`? why not move the disable/enable code into the condition in `handleHostgroupChangedNew`? Something like:
```javascript
function handleHostgroupChangedNew(element) {
  reset_explicit_values(element);
  set_inherited_value(element);
  // call for form update only if there is a hostgroup selected
  if ($('#host_hostgroup_id').val() != "") {
    $("#host_compute_resource_id").prop("disabled", true);
    update_form(element).then(function () {
      $("#host_compute_resource_id").prop("disabled", false);
    });
   }
 }
```